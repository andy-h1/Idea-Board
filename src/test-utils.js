import React from 'react';
import { render } from '@testing-library/react';
import { IdeaContext } from './contexts/IdeaContext';

const AllTheProviders = ({ children, customContext }) => (
  <IdeaContext.Provider
    value={{
      ideas: [],
      sortIdea: () => null,
      ...customContext
    }}
  >
    {children}
  </IdeaContext.Provider>
);

const customRender = (ui, customContext, options) => {
  /* eslint-disable react/display-name */
  return render(ui, {
    wrapper: () => <AllTheProviders customContext={customContext}>{ui}</AllTheProviders>,
    ...options
  });
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
