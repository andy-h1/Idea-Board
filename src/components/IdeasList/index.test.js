import React from 'react';
import userEvent from '@testing-library/user-event';
import { render } from '../../test-utils';
import { IdeasList } from '.';

describe('Ideas List Component', () => {
  it('should render', () => {
    const { container } = render(<IdeasList />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should call sortIdea with correct value', () => {
    // set up
    const ideas = [
      {
        title: 'Day Z',
        description: 'Survival',
        id: '1',
        time: new Date().toLocaleTimeString()
      }
    ];

    const sortIdea = jest.fn();

    // render
    const { getByTestId } = render(<IdeasList />, { ideas, sortIdea });
    const expectedSortValue = 'timeAsc';

    // faking the user behaviour i.e. clicking a dropdown
    userEvent.selectOptions(getByTestId('dropdown'), [expectedSortValue]);
    // assert - check it's called with what we expect
    expect(sortIdea).toHaveBeenCalledWith(expectedSortValue);
  });

  it.todo('should render message if no ideas');
  it.todo('should render the ideas when there are ideas');
});
