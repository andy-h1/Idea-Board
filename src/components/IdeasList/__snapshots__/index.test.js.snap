// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ideas List Component should render 1`] = `
<div
  class="sc-iqAbSa drSwsX"
>
  <p>
    An idea that is developed and put into action is more important than an idea that exists only as an idea...
  </p>
</div>
`;
